.github/workflows/unit-test.yml

```diff
@@ -187,6 +192,7 @@ jobs:
       - name: Run PennyLane Unit Tests
         if: steps.continue.outputs.confirm == 'true'
         env:
+          PYTHONWARNINGS: ${{ inputs.python_warning_level }} 
```

Comment:
In the `run` part of this step, you can just add the warning promotion using the `-W` CLI argument. Since we either have the argument as promote to errors or we don't have the argument at all, I would just get rid of this env var.

Reply:
Decided to go inline instead, as it will keep things localised to pytest.